reconcile/utils/terrascript_client.py

```diff
@@ -4073,7 +4093,63 @@ def populate_tf_resource_secrets_manager(self, resource, namespace_info):
 
         self.add_resources(account, tf_resources)
 
-    def populate_tf_resource_asg(self, resource, namespace_info):
+    def get_asg_image_id(self, image, existing_secrets, output_prefix,
+                         account, region):
+        image_id = image.get('id')
+        if image_id:
+            return image_id, ''
+
+        image_repo = image.get('repo')
+        tag_name = image_repo.get('tag_name')
+        url = image_repo.get('url')
+        ref = image_repo.get('ref')
+        pattern = r'^[0-9a-f]{40}$'
+        # get commit_sha from ref
+        if re.search(pattern, ref):
+            commit_sha = ref
+        # get commit_sha from branch
+        elif 'github' in url:
+            github = self.init_github()
+            repo_name = url.rstrip("/").replace('https://github.com/', '')
+            repo = github.get_repo(repo_name)
+            commit = repo.get_commit(sha=ref)
+            commit_sha = commit.sha
+        elif 'gitlab' in url:
+            raise ValueError("dose not support gitlab repo for now")
+
+        # Get the most recent AMI id
+        ci_account_name = self.ci_account['name']
+        owner = [self.ci_account['uid']]
+        filters = [{
+            'Name': 'tag:' + tag_name,
+            'Values': [commit_sha]
+        }]
+        image_id = self.aws.get_image_id(ci_account_name, region, owner,
+                                         filters)
+        if not image_id:
+            logging.warning(f"could not find ami with filters {filters}"
+                            f"in account {ci_account_name}, "
+                            f"use existing asg's ami info")
+            try:
+                existing_image = \
+                    existing_secrets[account][output_prefix]
+            except KeyError:
+                raise ValueError("could not find existing asg's ami info")
```

Comment:
Give more details here:
```suggestion
                raise ValueError(f"could not find existing asg's ami info for {account} with prefix {output_prefix}")
```
or something liek that